tests/test_check_stac_metadata_utils_logging.py

```diff
@@ -101,9 +103,10 @@ def should_log_non_s3_url_prefix_validation() -> None:
     with patch("geostore.check_stac_metadata.utils.LOGGER.error") as logger_mock, patch(
         "geostore.check_stac_metadata.utils.processing_assets_model_with_meta"
     ):
-        STACDatasetValidator(
-            hash_key, url_reader, MockAssetGarbageCollector(), MockValidationResultFactory()
-        ).run(metadata_url)
+        with raises(InvalidSTACRootTypeError):
```

Comment:
This `with` could just be part of the parent `with`, as in `should_raise_invalid_stack_root_type_error_for_non_collection_or_catalog`.

Reply:
Updated. Thanks.